community/server/src/main/java/org/neo4j/server/configuration/ConfigLoader.java

```diff
@@ -84,7 +91,7 @@ public Config loadConfig( Optional<File> configFile,
         config.ifPresent( ( c ) -> settings.putAll( loadFromFile( log, c ) ) );
         settings.putAll( toMap( configOverrides ) );
         overrideEmbeddedDefaults( settings );
-        settings.put( GraphDatabaseSettings.neo4j_home.name(), System.getProperty( "user.dir" ) );
+        homeDir.ifPresent( ( h ) -> settings.put( GraphDatabaseSettings.neo4j_home.name(), h.getAbsolutePath() ) );
```

Comment:
Wouldn't it be better to set this explicitly to `user.dir` if a value isn't provided?